monitoring/grafana/dashboards/jsonnet/grafana_dashboards.jsonnet

```diff
@@ -765,23 +765,32 @@ local addStyle(alias, colorMode, colors, dateFormat, decimals, mappingType, patt
         '$osd R/W Bytes', '', 'bytes', 'Read (-) / Write (+)', 'irate(ceph_osd_op_r_out_bytes{ceph_daemon=~\"$osd\"}[1m])', 'irate(ceph_osd_op_w_in_bytes{ceph_daemon=~\"$osd\"}[1m])', 'Read Bytes', 'Write Bytes', 12, 1, 6, 9)
       .addSeriesOverride({"alias": "Read Bytes","transform": "negative-Y"}),
-      addRowSchema(false, true, 'Physical Device Performance') + {gridPos: {x: 0, y: 10, w: 24, h: 1}},
+      addPieChartSchema({}, '$datasource', '', 'Under graph', 'pie', '$osd onode Hit, Miss Requests', 'current')
+      .addTarget(
+        addTargetSchema('ceph_bluestore_bluestore_onode_hits{ceph_daemon=~\"$osd\"}', 1, 'time_series', '$osd hits')
+      )
+      .addTarget(
+        addTargetSchema('ceph_bluestore_bluestore_onode_misses{ceph_daemon=~\"$osd\"}', 1, 'time_series', '$osd misses')
+      )  + {gridPos: {x: 20, y: 1, w: 6, h: 9}},
+      OsdDeviceDetailsPanel(
+        'Total onode hit/miss values', '', 'short', null, 'sum(ceph_bluestore_bluestore_onode_hits)', 'sum(ceph_bluestore_bluestore_onode_misses)', 'Hits', 'Misses', 0, 10, 24, 8),
+      addRowSchema(false, true, 'Physical Device Performance') + {gridPos: {x: 0, y: 20, w: 24, h: 1}},
       OsdDeviceDetailsPanel(
-        'Physical Device Latency for $osd', '', 's', 'Read (-) / Write (+)', '(label_replace(irate(node_disk_read_time_seconds_total[1m]) / irate(node_disk_reads_completed_total[1m]), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\") and on (instance, device) label_replace(label_replace(ceph_disk_occupation{ceph_daemon=~\"$osd\"}, \"device\", \"$1\", \"device\", \"/dev/(.*)\"), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\"))', '(label_replace(irate(node_disk_write_time_seconds_total[1m]) / irate(node_disk_writes_completed_total[1m]), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\") and on (instance, device) label_replace(label_replace(ceph_disk_occupation{ceph_daemon=~\"$osd\"}, \"device\", \"$1\", \"device\", \"/dev/(.*)\"), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\"))', '{{instance}}/{{device}} Reads', '{{instance}}/{{device}} Writes', 0, 11, 6, 9)
+        'Physical Device Latency for $osd', '', 's', 'Read (-) / Write (+)', '(label_replace(irate(node_disk_read_time_seconds_total[1m]) / irate(node_disk_reads_completed_total[1m]), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\") and on (instance, device) label_replace(label_replace(ceph_disk_occupation{ceph_daemon=~\"$osd\"}, \"device\", \"$1\", \"device\", \"/dev/(.*)\"), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\"))', '(label_replace(irate(node_disk_write_time_seconds_total[1m]) / irate(node_disk_writes_completed_total[1m]), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\") and on (instance, device) label_replace(label_replace(ceph_disk_occupation{ceph_daemon=~\"$osd\"}, \"device\", \"$1\", \"device\", \"/dev/(.*)\"), \"instance\", \"$1\", \"instance\", \"([^:.]*).*\"))', '{{instance}}/{{device}} Reads', '{{instance}}/{{device}} Writes', 0, 21, 6, 9)
```

Comment:
When introducing jsonnet/grafonnet, one of the key deciding factors was increasing maintainability/readability of the Grafana dashboard JSON code, so why not making things more readable? It's hard to find where a PromQL expression ends and the next starts (btw nested `label_replace()` should be a code smell if such a thing may exist in PromQL 🙈 ):

```suggestion
        'Physical Device Latency for $osd',
        '',
        's',
        'Read (-) / Write (+)',
        '(label_replace(irate(node_disk_read_time_seconds_total[1m]) / irate(node_disk_reads_completed_total[1m]),
  "instance", "$1", "instance", "([^:.]*).*") and on (instance, device) label_replace(  label_replace(ceph_disk_occupation{ceph_daemon=~"$osd"}, "device", "$1", "device", "/dev/(.*)"), "instance", "$1", "instance", "([^:.]*).*"))',
        '(label_replace(irate(node_disk_write_time_seconds_total[1m]) / irate(node_disk_writes_completed_total[1m]), "instance", "$1", "instance", "([^:.]*).*") and on (instance, device) label_replace(label_replace(ceph_disk_occupation{ceph_daemon=~"$osd"}, "device", "$1", "device", "/dev/(.*)"), "instance", "$1", "instance", "([^:.]*).*"))',
        '{{instance}}/{{device}} Reads', '{{instance}}/{{device}} Writes', 0, 21, 6, 9)
```

Reply:
Done. Thanks @epuertat